src/Vehicle/RemoteIDManager.cc

```diff
@@ -0,0 +1,424 @@
+
+    // Timer to send messages at a constant rate
+    _sendMessagesTimer.setInterval(SENDING_RATE_MSEC);
+    connect(&_sendMessagesTimer, &QTimer::timeout, this, &RemoteIDManager::_sendMessages);
+
+    // GCS GPS position updates to track the health of the GPS data
+    connect(_positionManager, &QGCPositionManager::positionInfoUpdated, this, &RemoteIDManager::_updateLastGCSPositionInfo);
+
+    // Check changes in basic id settings as long as they are modified
+    connect(_settings->basicID(), &Fact::rawValueChanged, this, &RemoteIDManager::_checkGCSBasicID);
+    connect(_settings->basicIDType(), &Fact::rawValueChanged, this, &RemoteIDManager::_checkGCSBasicID);
+    connect(_settings->basicIDUaType(), &Fact::rawValueChanged, this, &RemoteIDManager::_checkGCSBasicID);
+}
+
+void RemoteIDManager::mavlinkMessageReceived(mavlink_message_t& message )
+{
+    switch (message.msgid) {
+    // So far we are only listening to this one, as heartbeat won't be sent if connected by CAN
+    case MAVLINK_MSG_ID_OPEN_DRONE_ID_ARM_STATUS: 
+        _handleArmStatus(message);
+    default:
+        break;
+    }
+}
+
+// This slot will be called if we stop receiving heartbeats for more than RID_TIMEOUT seconds
+void RemoteIDManager::_odidTimeout()
+{
+    _commsGood = false;
+    _sendMessagesTimer.stop(); // We stop sending messages if the communication with the RID device is down
+    emit commsGoodChanged();
+    qCDebug(RemoteIDManagerLog) << "We stopped receiving heartbeat from RID device.";
+}
+
+// Parsing of the ARM_STATUS message comming from the RID device
+void RemoteIDManager::_handleArmStatus(mavlink_message_t& message)
+{
+    // Compid must be ODID_TXRX_X
+    if ( (message.compid < MAV_COMP_ID_ODID_TXRX_1) || (message.compid > MAV_COMP_ID_ODID_TXRX_3) ) {
+        // or same as autopilot, in the case of Ardupilot and CAN RID modules
+        if (message.compid != MAV_COMP_ID_AUTOPILOT1) {
+            return;
+        }
+    }
+
+    // Sanity check, only get messages from same sysid 
+    if (_vehicle->id() != message.sysid) {
+        return;
+    }
+
+    // We set the targetsystem
+    if (_targetSystem != message.sysid) {
+        _targetSystem = message.sysid;
+        qCDebug(RemoteIDManagerLog) << "Subscribing to ODID messages coming from system " << _targetSystem;
+    }
+
+    if (!_commsGood) {
+        _commsGood = true;
+        _sendMessagesTimer.start();     // Start sending our messages
+        _checkGCSBasicID();             // Check if basicID is good to send 
+        checkOperatorID();              // Check if OperatorID is good in case we want to send it from start because of the settings
+        emit commsGoodChanged();
+        qCDebug(RemoteIDManagerLog) << "Receiving ODID_ARM_STATUS from RID device";
+    }
+
+    // Restart the timeout
+    _odidTimeoutTimer.start();
+
+    // CompId and sysId are correct, we can proceed
+    mavlink_open_drone_id_arm_status_t armStatus;
```

Comment:
PX4 does not send this yet. Not blocking for this PR though.
I'm also not familiar what is required in detail for compliance.

Reply:
I am not an expert on this Remote ID thing, but according to what I learned during this implementation, talking with companies using it, RID device manufacturers, and Ardupilot team, it seems this message is what PX4 should send so the GCS understands the status of the vehicle regarding remote ID.

I am familiarized with Ardupilot but not PX4, so I can not help actively a lot on that from. But, when PX4 adds support for this I will be happy to help on the QGC side if needed. Also if anybody on PX4 team wants to ping me to clarify quickly any doubt they might have I am available anytime, probably the fastest way to reach me is by discord, I am in the dronecode servers, my nick is davidsastresas.

Thanks!

Reply:
Ok thanks for clarifying. It should be easy to add this one.